README.md

```diff
@@ -31,22 +40,55 @@ Name|Value Type|Example Value| Required|Default
 ----|----------|-------------|---------|-------------
 log-level|string|"warn"|false|"info"
 github-token|string| |true|null
-jira-user|string|"user@jira.example.com"|true|null
-jira-pass|string|"theUserPassword123"|true|null
+jira-user|string|"user@jira.example.com"|false|null
+jira-pass|string| |false|null
+jira-token|string| |false|null
+jira-secret|string| |false|null
+jira-consumer-key|string| |false|null
+jira-private-key|string| |false|null
 repo-name|string|"coreos/issue-sync"|true|null
 jira-uri|string|"https://jira.example.com|true|null
 jira-project|string|"SYNC"|true|null
 since|string|"2017-07-01T13:45:00-0800"|false|"1970-01-01T00:00:00+0000"
+timeout|duration|5000000000|false|60000000000
+
+### Configuration Key Descriptions
+
+`log-level` is the minimum level which will be logged; any output below
+this value will be discarded.
+
+`github-token` is a personal access token, used to access GitHub as a
+specific user.
+
+`jira-user` and `jira-pass` are the username (i.e. email) and password
+of the JIRA user which will be authenticated. See `Authentication` for
+more details.
+
+`jira-token` and `jira-secret` are OAuth access tokens which will be
+used to perform an OAuth connection to JIRA. `jira-consumer-key` and
+`jira-private-key` are the RSA key used for OAuth. See `Authentication`
+for more details.
 
-Note that the `repo-name` must include the owner and repo, that
-`jira-uri` is the base URI of the JIRA instance, that `jira-project` is
-the project's Key, not its name, and that `since` is in ISO-8601 format.
+`repo-name` is the GitHub repo from which issues will be retrieved. It
+must be in the form `owner/repo`, for example `coreos/issue-sync`.
 
-If the JIRA instance lives on a subdirectory of the server, that must
-be included in the jira-uri. For example, at "https://example.com/jira".
+`jira-uri` is the base URI of the JIRA instance. If the JIRA instance
+lives on a subdirectory, that must be included. For example,
```

Comment:
In HTTP these are not subdirectories, since it does not necessarily represent a file-system. Consider writing:
If the JIRA instance lives at a non-root URL, the path must be included in the URL.